halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/providers/azure/AzureAddAccountCommand.java

```diff
@@ -101,6 +101,12 @@ protected String getProviderName() {
   )
   private List<String> regions = new ArrayList<String>(Arrays.asList("westus", "eastus"));
 
+  @Parameter(
+      names = "--useSshPublicKey",
+      description = AzureCommandProperties.USE_SSH_PUBLIC_KEY_DESCRIPTION
+  )
+  private String useSshPublicKey = "true";
```

Comment:
Any reason not to make this a `Boolean`?

Reply:
I tried Boolean at first but it did not work. The effect of Boolean was the hal-cli is able to recognize this newly added parameter (add the parameter to the /.hal/config file) but unable to get its followed true/false value. Changing the type to String worked. I struggled for a while but with no luck though I did notice there are other hal-cli parameters having Boolean type.

One possible reason might be due to my testing method. To test the hal change locally, my way was replacing those .class files in the official hal-X.jar(s) with my changed .class files, and then run "hal ...". I'm not aware of other testing methods. Though there was ever a suggestion to run "./hal ..." under the hal-cli folder, its prerequisite is all changes have to happen within hal-cli. But my case here also changed hal-config so that way does not work. Please possibly suggest if there is any formal testing method to locally test local hal change.

Is it possible to accept this first? Thus we could quickly unblock something whose work around is having to prepare some long instructions telling how to manually enable this feature in clouddriver. We could make a following up PR to update the parameter type if we could find a way later.

Reply:
Sure, sounds good. Will get this merged and then comment on how to develop on hal locally.

Reply:
I test hal changes locally as follows:

* Open 2 terminal windows.
* Clone this repo.
* Start the halyard daemon via `./gradlew halyard-web:run` (from the `halyard` directory).
* Build the hal cli via `../gradlew installDist` (from the `halyard/halyard-cli` directory).
* Run `./hal` commands (from the `halyard/halyard-cli` directory).

Have to restart the daemon if I change classes the daemon depends on, and rebuild the cli if I change classes it depends on.